warehouse/query-core/src/main/java/datawave/query/transformer/ExcerptTransform.java

```diff
@@ -84,12 +91,111 @@ public Entry<Key,Document> apply(@Nullable Entry<Key,Document> entry) {
+        }
+        // now lets find all of the terms for excerpt fields and add them to the list
+        if (document.containsKey(JexlEvaluation.HIT_TERM_FIELD)) {
+            Attributes hitList = (Attributes) document.get(JexlEvaluation.HIT_TERM_FIELD);
+            // for each hit term
+            for (Attribute<?> attr : hitList.getAttributes()) {
+                ValueTuple hitTuple = attributeToHitTuple(attr);
+                // if this is for a requested excerpt field
+                if (excerptFields.containsField(hitTuple.getFieldName())) {
+                    // get the offset, preferring offsets that overlap with existing phrases for this field/eventId
+                    TermWeightPosition pos = getOffset(hitTuple, phraseIndexMap);
+                    if (pos != null) {
+                        // add the term as a phrase as defined in the term weight position. Note that this will collapse with any overlapping phrases already in
+                        // the list.
+                        allPhraseIndexes.addIndexTriplet(String.valueOf(hitTuple.getFieldName()), keyToEventId(attr.getMetadata()), pos.getLowOffset(),
+                                        pos.getOffset());
+                    }
+                }
+            }
+        }
+        // return the file set of phrase indexes if any
+        return allPhraseIndexes;
+    }
+    
+    /**
+     * Get the term weight position (offset) for the specified hit term. This will return an offset overlapping a phrase in the existing phrase index map first.
+     * Otherwise the first position will be returned.
+     * 
+     * @param hitTuple
+     *            The hit term tuple
+     * @param phraseIndexMap
+     *            The phrase indexes in map form to speed up searches.
+     * @return The TermWeightPosition for the given hit term
+     */
+    private TermWeightPosition getOffset(ValueTuple hitTuple, Map<String,Map<String,PhraseIndexes>> phraseIndexMap) {
+        Key docKey = hitTuple.getSource().getMetadata();
+        // if we do not know the source document key, then we cannot find the term offset
+        if (docKey == null) {
+            log.warn("Unable to find the source document for a hit term, skipping excerpt generation");
             return null;
         }
+        String fieldName = hitTuple.getFieldName();
+        String eventId = keyToEventId(docKey);
+        
+        // get the key at which we would find the term frequencies
+        Key tfKey = new Key(docKey.getRow().toString(), Constants.TERM_FREQUENCY_COLUMN_FAMILY.toString(), docKey.getColumnFamily().toString() + '\u0000'
+                        + hitTuple.getValue() + '\u0000' + hitTuple.getFieldName());
+        Range range = new Range(tfKey, tfKey.followingKey(PartialKey.ROW_COLFAM_COLQUAL));
+        try {
+            // seek directly to that key
+            source.seek(range, Collections.emptyList(), false);
+            if (source.hasTop()) {
```

Comment:
I'm wondering if we need to build in some limits somewhere. I'm thinking of a case where we have an incredibly long document with many term positions and we've searched that appears incredibly frequently in that document. Perhaps the phrase index map provides sufficient limiting here or since we are returning only the first offset of the term is is OK. Do we create a circuit breaker that will bail after we've visited some number of tf's (This is probably not the right place to do this just thought of it while looking at this code).